File: rails/app/models/location_account.rb
# == Schema Information
#
# Table name: location_accounts
#
#  approved_by_location_at :datetime
#  approved_by_user_at     :datetime
#  attendance_status       :text
#  permission_level        :text
#  role                    :text             not null
#  title                   :text
#  created_at              :datetime         not null
#  updated_at              :datetime         not null
#  external_id             :text
#  location_id             :uuid             not null
#  user_id                 :uuid             not null
#
# Indexes
#
#  index_location_accounts_on_external_id  (external_id) UNIQUE
#  index_location_accounts_on_location_id  (location_id)
#  index_location_accounts_on_role         (role)
#  index_location_accounts_on_user_id      (user_id)
#
# Foreign Keys
#
#  fk_rails_...  (location_id => locations.id) ON DELETE => cascade
#  fk_rails_...  (user_id => users.id) ON DELETE => cascade
#
class LocationAccount < ApplicationRecord
  belongs_to :user
  belongs_to :location
end

File: mobile/src/common/models/LocationAccount.ts
import { User } from './User'
import { Location } from './Location'
import { Model, attribute as attr, initialize, relationship, STRING } from './Model'

export class LocationAccount extends Model {
  static singular = 'locationAccount'
  static plural = 'locationAccounts'

  constructor(data?: any) {
    super()
    initialize(this, data)
  }

  @attr({ type: STRING }) locationId: string | null = null
  @attr({ type: STRING }) externalId: string | null = null
  @attr({ type: STRING }) role: string | null = null
  @attr({ type: STRING }) title: string | null = null
  @attr({ type: STRING }) permissionLevel: string | null = null
  @attr({ type: STRING }) attendanceStatus: string | null = null


  @relationship({ type: 'hasOne', model: 'location' })
  location: Location = new Location()

  @relationship({ type: 'hasOne', model: 'user' })
  user: User = new User()
}
